seatunnel-connectors-v2/connector-flink-table-store/src/main/java/org/apache/seatunnel/connectors/seatunnel/fts/source/FlinkTableStoreSource.java

```diff
@@ -0,0 +1,117 @@
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.fts.source;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+
+import org.apache.seatunnel.api.common.PrepareFailException;
+import org.apache.seatunnel.api.common.SeaTunnelAPIErrorCode;
+import org.apache.seatunnel.api.source.Boundedness;
+import org.apache.seatunnel.api.source.SeaTunnelSource;
+import org.apache.seatunnel.api.source.SourceReader;
+import org.apache.seatunnel.api.source.SourceSplitEnumerator;
+import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.common.config.CheckConfigUtil;
+import org.apache.seatunnel.common.config.CheckResult;
+import org.apache.seatunnel.common.constants.PluginType;
+import org.apache.seatunnel.connectors.seatunnel.fts.exception.FlinkTableStoreConnectorException;
+import org.apache.seatunnel.connectors.seatunnel.fts.utils.RowTypeConverter;
+
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.table.store.table.FileStoreTableFactory;
+import org.apache.flink.table.store.table.Table;
+
+import com.google.auto.service.AutoService;
+
+import static org.apache.seatunnel.connectors.seatunnel.fts.config.FlinkTableStoreConfig.TABLE_PATH;
+
+@AutoService(SeaTunnelSource.class)
+public class FlinkTableStoreSource
+        implements SeaTunnelSource<
+                SeaTunnelRow, FlinkTableStoreSourceSplit, FlinkTableStoreSourceState> {
+
+    private static final long serialVersionUID = 1L;
+
+    public static final String PLUGIN_NAME = "FlinkTableStore";
+
+    private Config pluginConfig;
+
+    private SeaTunnelRowType seaTunnelRowType;
+
+    private Table table;
+
+    @Override
+    public String getPluginName() {
+        return PLUGIN_NAME;
+    }
+
+    @Override
+    public void prepare(Config pluginConfig) throws PrepareFailException {
+        this.pluginConfig = pluginConfig;
+        CheckResult result = CheckConfigUtil.checkAllExists(pluginConfig, TABLE_PATH.key());
+        if (!result.isSuccess()) {
+            throw new FlinkTableStoreConnectorException(
+                    SeaTunnelAPIErrorCode.CONFIG_VALIDATION_FAILED,
+                    String.format(
+                            "PluginName: %s, PluginType: %s, Message: %s",
+                            getPluginName(), PluginType.SOURCE, result.getMsg()));
+        }
+        // initialize flink table store
+        String tablePath = pluginConfig.getString(TABLE_PATH.key());
+        table = FileStoreTableFactory.create(new Path(tablePath));
+
+        // initialize seatunnel row type
+        // TODO: Support column projection
+        seaTunnelRowType = RowTypeConverter.convert(table.rowType());
+    }
+
+    @Override
+    public Boundedness getBoundedness() {
+        return Boundedness.BOUNDED;
```

Comment:
Why `Boundedness.BOUNDED`?

Reply:
Now I only implement the batch mode to read paimon table, so it returns Boundedness.BOUNDED